src/main/java/io/quarkus/gizmo2/creator/TypeCreator.java

```diff
@@ -102,6 +101,7 @@ default void implements_(Class<?> interface_) {
         if (! interface_.isInterface()) {
             throw new IllegalArgumentException("Only interfaces may be implemented");
         }
+        implements_(ClassDesc.of(interface_.getName()));
```

Comment:
Heh, oops!

Reply:
For cases like this you should use `Util.classDesc(interface_)` btw. It caches the descriptor for each type.